### PR TITLE
[SPARK-31570][R][DOCS] R combine gapply dapply

### PR DESCRIPTION
**Pull request by MichaelChirico**
### What changes were proposed in this pull request?

Combine `gapply`, `gapplyCollect`, `dapply`, and `dapplyCollect` in a single Rd. This will help simplify changes to the highly correlated documentation as well as make it easier for users to see related functions in a single location.

### Why are the changes needed?

https://issues.apache.org/jira/browse/SPARK-31570

### Does this PR introduce any user-facing change?

Nothing in code, only in documentation

### How was this patch tested?

Ran `roxygen2::roxygenise('.')` and examined the Rd output

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by MichaelChirico** [comment]
@HyukjinKwon just merged vs master, ready to merge when you are 👍

**Comment by HyukjinKwon** [comment]
retest this please

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1652,34 +1607,43 @@ setMethod("dapplyCollect",
 
 #' gapply
```
Comment (HyukjinKwon) [inline R/pkg/R/DataFrame.R:1608]: Shall we update here too? Seems a bit odd that the short summary says gapply for dapply ones:
Screen Shot 2020-05-19 at 6 16 03 PM
Screen Shot 2020-05-19 at 6 16 15 PM

Comment (HyukjinKwon) [reply R/pkg/R/DataFrame.R:1608]: We could change something like "R native function execution". There are a lot of examples like this to group it in `functions.R`. Might be best to follow the way consistently.

Comment (HyukjinKwon) [reply R/pkg/R/DataFrame.R:1608]: You can check the documentation build via `cd R && ./create-docs.sh && open pkg/html/index.html`

Comment (MichaelChirico) [reply R/pkg/R/DataFrame.R:1608]: Thanks @HyukjinKwon , i went with `Functions to apply native R code on groups or partitions`. 

Generally I think the Rd titles could go for a fresh pass overall (e.g. `cov` is not very descriptive and applies to a bevy of functions), but I'll leave that for another day.

**Comment by MichaelChirico** [comment]
Error seems related to SBT, not this PR

**Comment by MichaelChirico** [comment]
@HyukjinKwon could you help to revive this PR?

**Pull request closed without merging**